src/bin/trace-verifier/utils.rs

```diff
@@ -55,31 +58,34 @@ pub fn verify(
         ));
         let file = std::fs::File::create(&path).unwrap();
         report.flamegraph(file).unwrap();
-        info!("Profiling report saved to: {:?}", path);
+        dev_info!("Profiling report saved to: {:?}", path);
     }
 
     let elapsed = now.elapsed();
 
     if root_after != revm_root_after {
-        error!("Root after in trace: {:x}", root_after);
-        error!("Root after in revm: {:x}", revm_root_after);
-        error!("Root mismatch");
+        dev_error!("Root after in trace: {:x}", root_after);
+        dev_error!("Root after in revm: {:x}", revm_root_after);
+        dev_error!("Root mismatch");
+
         if !log_error {
             std::process::exit(1);
         }
-        return false;
+        return Ok(false);
```

Comment:
return err instead of `Ok(false)`:
```suggestion
        return Err(VerificationError::StateRootMismatch(root_after, revm_root_after));
```